jobs/build-charms/charms.py

```diff
@@ -681,19 +696,11 @@ def charm_build(self):
                 args += " --charm-file"
                 self.dst_path = str(Path(self.src_path) / f"{self.name}.charm")
             self.echo(f"Building with: charm build {args}")
-            try:
-                _CharmStore(self).build(*args.split(), _cwd=self.src_path)
-            except sh.ErrorReturnCode:
-                ret.ok = False
-
+            ret = _CharmStore(self).build(*args.split(), _cwd=self.src_path)
```

Comment:
is _CharmStore used to build legacy charms for both the charm store and the hub?

Reply:
this is obviously unclear.  
The command `charm` is used to build legacy charms (charms with a layer.yaml) regardless of which "store" they end up in.  if the target store is "charmhub" then you have to add `--charm-file`.  I wonder if i can rework this.  I'll test if `charmcraft` can build legacy charms now.  

The command `charmcraft` is used to build operator charms (newest version)

Reply:
yeah, `charmcraft` wasn't able to build a reactive charm.  But i COULD update this function to look like this:

What do you think @stonepreston 

```python
    def charm_build(self):
        """Perform a build against charm/bundle."""
        if "override-build" in self.opts:
            self.echo("Override build found, running in place of charm build.")
            ret = script(
                self.opts["override-build"],
                cwd=self.src_path,
                charm=self.name,
                echo=self.echo,
            )
        elif self.legacy_charm:
            args = "-r --force -i https://localhost"
            if self.store == "ch":
                args += " --charm-file"
                self.dst_path = str(Path(self.src_path) / f"{self.name}.charm")
            self.echo(f"Building with: charm build {args}")
            ret = CharmCmd(self).build(*args.split(), _cwd=self.src_path)
        else:
            args = f"-f {self.src_path}"
            self.echo(f"Building with: charmcraft build {args}")
            ret = CharmcraftCmd(self).build(*args.split(), _cwd=self.build.build_dir)
        if not ret.ok:
            self.echo("Failed to build, aborting")
            raise SystemExit(f"Failed to build {self.name}")
```

Reply:
I think that's much clearer